linalg/densemat.hpp

```diff
@@ -768,9 +775,9 @@ class DenseTensor
 
    int TotalSize() const { return SizeI()*SizeJ()*SizeK(); }
 
-   void SetSize(int i, int j, int k)
+   void SetSize(int i, int j, int k, MemoryType mt_ = MemoryType::SIZE)
```

Comment:
I think it is better for clarity to add a new constant in `MemoryType`, say `MemoryType::PRESERVE` which can be used only in certain contexts like this.

Reply:
Yeah that makes sense, we've been talking about that in GH-1669 as well. I like `PRESERVE`, I'll add that